File: src/vta/instruction_queue.cc
#include "vta/instruction_queue.hh"

#include "base/trace.hh"
#include "debug/BaseVTAFlag.hh"

namespace gem5
{
InstructionQueue::InstructionQueue(const Params &params) :
    SimObject(params),
    pushLatency(params.push_latency),
    popLatency(params.pop_latency)
{
    DPRINTF(BaseVTAFlag, "create InstructionQueue!\n");
}

} // namespace gem5

File: src/vta/InstructionQueue.py
from m5.params import *
from m5.SimObject import SimObject


class InstructionQueue(SimObject):
    type = "InstructionQueue"
    cxx_header = "vta/instruction_queue.hh"
    cxx_class = "gem5::InstructionQueue"

    push_latency = Param.Cycles(10, "Cycles to wait push")
    pop_latency = Param.Cycles(10, "Cycles to wait pop")
